src/status_im/components/drawer/view.cljs

```diff
@@ -34,60 +34,70 @@
+            valid-name?  [:my-profile.drawer/valid-name?]
+            placeholder  [:get :my-profile/default-name]]
     [view st/name-input-wrapper
      [text-input
       {:placeholder    placeholder
-       :style          (st/name-input-text (s/valid? ::profile.db/name @name-text))
+       :style          (st/name-input-text valid-name?)
        :font           :medium
-       :default-value  (or @name-text previous-name)
-       :on-change-text #(reset! name-text %)
-       :on-end-editing #(if (s/valid? ::profile.db/name @name-text)
-                          (rf/dispatch [:account-update {:name (utils/clean-text (or @name-text placeholder))}])
-                          (reset! name-text previous-name))}]]))
+       :default-value profile-name
+       :on-focus       #(rf/dispatch [:my-profile.drawer/edit-name])
+       :on-change-text #(rf/dispatch [:my-profile.drawer/update-name %])
+       :on-end-editing #(rf/dispatch [:my-profile.drawer/save-name])}]]))
 
 (defview status-input []
-  [account      [:get-current-account]
-   status-edit? (r/atom false)
-   status-text  (r/atom nil)]
-  (let [placeholder     (i18n/label :t/update-status)
-        previous-status (:status account)]
-    [view st/status-container
-     (if @status-edit?
-       [text-input {:style               st/status-input-view
-                    :multiline           true
-                    :auto-focus          true
-                    :focus               @status-edit?
-                    :max-length          140
-                    :accessibility-label id/drawer-status-input
-                    :placeholder         placeholder
-                    :default-value       previous-status
-                    :on-change-text      #(let [new-status (utils/clean-text %)]
-                                            (reset! status-text new-status)
-                                            (when (str/includes? % "\n")
-                                              (reset! status-edit? false)
-                                              (rf/dispatch [:my-profile/update-status new-status])))
-                    :on-blur             #(do (reset! status-edit? false)
-                                              (rf/dispatch [:my-profile/update-status @status-text]))}]
-
-       [status-view/status-view {:style           (st/status-view (str/blank? previous-status))
-                                 :on-press        #(reset! status-edit? true)
-                                 :number-of-lines 3
-                                 :status          (if (str/blank? previous-status)
-                                                    placeholder
-                                                    previous-status)}])]))
+
+  (letsubs [edit-status? [:my-profile.drawer/get :edit-status?]
+            status       [:my-profile.drawer/get :status]]
+    (let [placeholder (i18n/label :t/update-status)]
+      [view st/status-container
+       (if edit-status?
```

Comment:
I'd consider breaking these two vectors out into util fns so if statement is clear, no need for newline then